provider/cloudflare/cloudflare.go

```diff
@@ -191,13 +198,52 @@ func updateDataLocalizationRegionalHostnameParams(cfc cloudFlareChange) cloudfla
 
 // getCreateDNSRecordParam is a function that returns the appropriate Record Param based on the cloudFlareChange passed in
 func getCreateDNSRecordParam(cfc cloudFlareChange) cloudflare.CreateDNSRecordParams {
-	return cloudflare.CreateDNSRecordParams{
+	recordParam := cloudflare.CreateDNSRecordParams{
 		Name:    cfc.ResourceRecord.Name,
 		TTL:     cfc.ResourceRecord.TTL,
 		Proxied: cfc.ResourceRecord.Proxied,
 		Type:    cfc.ResourceRecord.Type,
-		Content: cfc.ResourceRecord.Content,
 	}
+
+	if cfc.ResourceRecord.Type == "SRV" {
+		recordParam.Data = cfc.ResourceRecord.Data
+	} else {
+		recordParam.Content = cfc.ResourceRecord.Content
+	}
+
+	return recordParam
+}
+
+// parseSRVContent parses the SRV record content string into the structured data required by the Cloudflare API
+func parseSRVContent(content string) (map[string]interface{}, error) {
```

Comment:
Error cases not covered, could we add unit-test coverege for them? We are aiming to increase test-coverage

![Screenshot 2025-02-27 at 07 21 33](https://github.com/user-attachments/assets/eafebfa2-9902-4324-9818-9e36c45bcf9d)